test/SpagoSpec.hs

```diff
@@ -92,6 +91,11 @@ spec = beforeAll clean $ around_ (withCwd testDir) $ do
       writeTextFile "packages.dhall" "let pkgs = ./packagesBase.dhall in pkgs // { simple-json = pkgs.simple-json // { version = \"d45590f493d68baae174b2d3062d502c0cc4c265\" } }"
       spago ["install", "simple-json"] >>= shouldBeSuccess
 
+    it "Spago should be able to install a package version by branch name with `/`'s in it" $ do
+
+      writeTextFile "packages.dhall" "let pkgs = ./packagesBase.dhall in pkgs // { simple-json = pkgs.simple-json // { version = \"compiler/0.12\" } }"
```

Comment:
Not sure if this is the best way to handle/test this since this branch could be removed at any time.

Reply:
Agreed, let's not depend on some random branches. I created [this properly named branch](https://github.com/spacchetti/purescript-metadata/tree/spago-test/branch-with-slash) in this fake package, and I promise I won't remove it 😄 

Note that the `metadata` package is already in the set so you can override the version in the same way